# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1481310212 18000
#      Fri Dec 09 14:03:32 2016 -0500
# Branch stable
# Node ID 233fb529184b1b083c51e599d341928abe2caf06
# Parent  3bc6e9a71d9e5c5a53549785ad49b2789129a232
exit with non-zero status if parsing or evaluation of --eval code fails

* interpreter.cc (interpreter::execute_eval_option_code):
Set parse_status instead of printing error message if
octave::execution_exception is caught.

diff --git a/libinterp/corefcn/interpreter.cc b/libinterp/corefcn/interpreter.cc
--- a/libinterp/corefcn/interpreter.cc
+++ b/libinterp/corefcn/interpreter.cc
@@ -797,18 +797,17 @@ namespace octave
     catch (const octave::interrupt_exception&)
       {
         recover_from_exception ();
       }
     catch (const octave::execution_exception&)
       {
         recover_from_exception ();
 
-        std::cerr << "error: unhandled execution exception -- eval failed"
-                  << std::endl;
+        parse_status = 1;
       }
 
     return parse_status;
   }
 
   void interpreter::execute_command_line_file (const std::string& fname)
   {
     octave::unwind_protect frame;
